lib/improver/tests/nowcasting/pysteps_advection/test_PystepsExtrapolate.py

```diff
@@ -0,0 +1,242 @@
+            cube.coord(axis='x').points = xpoints
+            cube.coord(axis='y').points = ypoints
+
+    def test_basic(self):
+        """Test output is a list of cubes with expected contents and
+        global attributes"""
+        expected_analysis = self.rain_cube.data.copy()
+        result = self.plugin.process(
+            self.rain_cube, self.ucube, self.vcube, self.orogenh_cube)
+        self.assertIsInstance(result, list)
+        # check result is a list including a cube at the analysis time
+        self.assertEqual(len(result), 9)
+        self.assertIsInstance(result[0], iris.cube.Cube)
+        self.assertIsInstance(result[0].data, np.ma.MaskedArray)
+        self.assertEqual(result[0].data.dtype, np.float32)
+        self.assertArrayAlmostEqual(result[0].data, expected_analysis)
+        # check for expected attributes
+        self.assertEqual(result[0].attributes['source'], 'MONOW')
+        self.assertEqual(
+            result[0].attributes['title'],
+            'MONOW Extrapolation Nowcast on UK 2 km Standard Grid')
+
+    def test_time_coordinates(self):
+        """Test cubelist has correct time metadata"""
+        result = self.plugin.process(
+            self.rain_cube, self.ucube, self.vcube, self.orogenh_cube)
+        for i, cube in enumerate(result):
+            # check values (and implicitly units - all seconds)
+            tdiff_seconds = i*self.interval*60
+            self.assertEqual(cube.coord('forecast_reference_time').points[0],
+                             self.rain_cube.coord('time').points[0])
+            self.assertEqual(
+                cube.coord('forecast_period').points[0], tdiff_seconds)
+            self.assertEqual(
+                cube.coord('time').points[0],
+                self.rain_cube.coord('time').points[0] + tdiff_seconds)
+
+            # check datatypes
+            self.assertEqual(cube.coord('time').dtype, np.int64)
+            self.assertEqual(
+                cube.coord('forecast_reference_time').dtype, np.int64)
+            self.assertEqual(cube.coord('forecast_period').dtype, np.int32)
+
+    def test_existing_time_coordinates_respected(self):
+        """Test that time coordinates are not added to a cube that already
+        has them"""
+        expected_coords = {
+            'projection_y_coordinate', 'projection_x_coordinate',
+            'forecast_period', 'forecast_reference_time', 'time'}
+        frt_coord = self.rain_cube.coord('time').copy()
+        frt_coord.rename('forecast_reference_time')
+        self.rain_cube.add_aux_coord(frt_coord)
+        self.rain_cube.add_aux_coord(
+            iris.coords.AuxCoord(np.array([0], dtype=np.int32),
+                                 'forecast_period', 'seconds'))
+        result = self.plugin.process(
+            self.rain_cube, self.ucube, self.vcube, self.orogenh_cube)
+        result_coords = {coord.name() for coord in result[0].coords()}
```

Comment:
Why does this use `{}` instead of `[]`?

Reply:
Because I don't really care what order the coordinates are returned in - I just care that they're all there.

Reply:
I've learned that this is "set comprehension" rather than "list comprehension"